lib/periodicChecks.js

```diff
@@ -0,0 +1,213 @@
+    })
+  );
+};
+
+/**
+ * This function assigns the PR author to a pull request.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const assignPrAuthor = async (context, pullRequest) => {
+  context.github.issues.addAssignees(
+    context.repo({
+      issue_number: pullRequest.number,
+      assignees: [pullRequest.user.login],
+    })
+  );
+};
+
+/**
+ * This function handles when a pull request has been approved.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const handleApproval = async (context, pullRequest) => {
+  // Check if author can merge PR. All members can merge the pull request.
+  const membershipCheckResponse = await context.github.orgs.checkMembership({
+    org: 'oppia',
+    username: pullRequest.user.login,
+  });
+  const SUCCESS_STATUS = 204;
+
+  if (membershipCheckResponse.status === SUCCESS_STATUS) {
+    // Assign author.
+    await assignPrAuthor(context, pullRequest);
+    await context.github.issues.createComment(
+      context.repo({
+        issue_number: pullRequest.number,
+        body:
+          'Hi @' +
+          pullRequest.user.login +
+          ', this PR is ready to be merged. Please make sure there are no ' +
+          'pending comments before merge. Thanks!',
+      })
+    );
+  } else {
+    // Assign project owner.
+    const label = getChangelogLabelFromPullRequest(pullRequest);
+    if (label) {
+      const projectOwner = getProjectOwnerFromLabel(label);
+      await context.github.issues.addAssignees(
+        context.repo({
+          issue_number: pullRequest.number,
+          assignees: [projectOwner],
+        })
+      );
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body:
+            'Hi @' +
+            projectOwner +
+            ', this PR is ready to be merged. Please make sure there are no ' +
+            "pending comments from the author's end before merge. Thanks!",
+        })
+      );
+    } else {
+      await assignPrAuthor(context, pullRequest);
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body:
+            'Hi @' +
+            pullRequest.user.login +
+            ', this PR is ready to be merged. Please ask one of the ' +
+            'reviewers to help with the merge, also, make sure there are no ' +
+            'pending comments before merge. Thanks!',
+        })
+      );
+    }
+  }
+};
+
+/**
+ * This function ensures that a pull request is assigned.
+ * 1. Fetch all pull requests.
+ * 2. Filter out those without assignees.
+ * 3. Find out appropriate assignee.
+ * 4. Assign appropriate assignee.
+ * 5. Ping appropriate assignee.
+ * @param {import('probot').Context} context
+ */
+const ensurePullRequestIsAssigned = async (context) => {
```

Comment:
Can we do one other thing here - log the assignees/reviewers who haven't responded within 24 hours of being assigned?